### PR TITLE
NSSet containsObject should return NO if the object is nil.

### LINKED ISSUE #498: NSSet containsObject should return NO for nil object.

### PR DESCRIPTION
**Pull request by RG-J**
fix #498

### CODE REVIEW

File: Frameworks/Foundation/NSSet.mm [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
-#import "Foundation/NSMutableSet.h"
 #import "Foundation/NSCountedSet.h"
 #import "Foundation/NSEnumerator.h"
-#import "NSEnumeratorInternal.h"
 #import "Foundation/NSKeyedArchiver.h"
+#import "Foundation/NSMutableSet.h"
+#import "NSEnumeratorInternal.h"
 #import "NSKeyedArchiverInternal.h"
+#import "Starboard.h"
+#import "StubReturn.h"
```
Comment (rajsesh-msft) [inline Frameworks/Foundation/NSSet.mm]: please revert the changes to the header file ordering.

**Comment by rajsesh-msft** [comment]
@ms-jihua is added to the review. #Closed

**Comment by rajsesh-msft** [comment]
@bbowman is added to the review. #Closed

File: Frameworks/Foundation/NSSet.mm [hunk 2]
```diff
@@ -370,6 +370,9 @@ + (NSSet*)set {
  @Status Interoperable
 */
 - (BOOL)containsObject:(id)obj {
+    if (obj == nil) {
+        return NO;
+    }
```
Comment (rajsesh-msft) [inline Frameworks/Foundation/NSSet.mm]: please move this to member method instead. We are getting a new version of this class and that will make merging this easier.

**Comment by rajsesh-msft** [comment]
:clock1:

**Comment by RG-J** [comment]
@rajsesh-msft nil issue addressed, please take a look.

**Comment by jaredhms** [comment]
:shipit:

File: Frameworks/Foundation/NSSet.mm [hunk 3]
```diff
@@ -526,6 +526,9 @@ - (NSArray*)sortedArrayUsingDescriptors:(NSArray*)descriptors {
  @Status Interoperable
 */
 - (id)member:(id)obj {
+    if (obj == nil) {
+        return nil;
+    }
```
Comment (bbowman) [inline Frameworks/Foundation/NSSet.mm]: Insufficient. You should make sure that nil can't be added to the set. That more closely matches the reference platform and then this check would not be needed.

**Comment by bbowman** [comment]
:clock1:

File: Frameworks/Foundation/NSSet.mm (hunk 3 above)
Comment (bbowman) [inline Frameworks/Foundation/NSSet.mm]: discussed offline. we still want this check and ideally would like a nil check on adding a member in the mutable version. The change, as is, is good though.

---
In reply to: [63953430](https://github.com/Microsoft/WinObjC/pull/499#discussion_r63953430) [](ancestors = 63953430)
